File: backend/src/main/java/ru/ngtu/sabacc/game/GameSessionFactory.java
package ru.ngtu.sabacc.game;

import ru.ngtu.sabacc.game.messaging.IGameMessageExchanger;
import ru.ngtu.sabacc.game.messaging.IGameSession;

/**
 * @author Egor Bokov
 */
public interface GameSessionFactory {
    //TODO add required init parameters
    IGameSession createSession(IGameMessageExchanger messageExchanger, Long sessionId);
}

File: backend/src/main/kotlin/ru/ngtu/sabacc/gamecore/gamesession/GameSessionFactory.kt
package ru.ngtu.sabacc.gamecore.gamesession

import ru.ngtu.sabacc.game.GameSessionFactory
import ru.ngtu.sabacc.game.messaging.IGameMessageExchanger
import ru.ngtu.sabacc.game.messaging.IGameSession

class GameSessionFactory : GameSessionFactory {
    override fun createSession(messageExchanger: IGameMessageExchanger, sessionId: Long): IGameSession {
        return GameSession(sessionId, messageExchanger)
    }
}
